src/equations.jl

```diff
@@ -48,6 +48,13 @@ Base.:~(lhs::Number    , rhs::Num) = Equation(value(lhs), value(rhs))
 Base.:~(lhs::Symbolic, rhs::Symbolic) = Equation(value(lhs), value(rhs))
 Base.:~(lhs::Symbolic, rhs::Any    ) = Equation(value(lhs), value(rhs))
 Base.:~(lhs::Any, rhs::Symbolic    ) = Equation(value(lhs), value(rhs))
+for T in [:Num, :Complex, :Number], S in [:Num, :Complex, :Number]
+    (T != :Complex && S != :Complex) && continue
+    @eval Base.:~(a::$T, b::$S) = [
+      (isa(value(real(a)), Number) && isa(value(real(b)), Number)) ? [] : value(real(a)) ~ value(real(b));
```

Comment:
This check is a bit inefficient and hard to read. You should probably check each case and return an `Equation` instead of a `Vector` if possible. Also, `isa` is an binary operator, so you can write `a isa Number`.

Reply:
I've tried to improve the code a bit. Is this better? Honestly the reason I did it like this is because "statement" thinking in this "inline" function definition. But of course Julia is expression based so you can just throw "statements" in there.